lib/icalendar/util/deserialize.ex

```diff
@@ -12,29 +13,66 @@ defmodule ICalendar.Util.Deserialize do
   end
 
   @doc ~S"""
-  iex> ICalendar.Util.Deserialize.retrieve_kvs("lorem:ipsum")
-  {"LOREM", "ipsum"}
+  This function extracts the key and value parts from each line of a iCalendar
+  string.
+
+      iex> ICalendar.Util.Deserialize.retrieve_kvs("lorem:ipsum")
+      %ICalendar.Property{key: "LOREM", params: %{}, value: "ipsum"}
   """
   def retrieve_kvs(line) do
+    # Split Line up into key and value
     [key, value] = String.split(line, ":", parts: 2, trim: true)
-    {String.upcase(key), value}
+    [key, params] = retrieve_params(key)
+
+    %Property{key: String.upcase(key), value: value, params: params}
+  end
+
+  @doc ~S"""
+  This function extracts parameter data from a key in an iCalendar string.
+
+      iex> ICalendar.Util.Deserialize.retrieve_params(
+      ...>   "DTSTART;TZID=America/Chicago")
+      ["DTSTART", %{"TZID" => "America/Chicago"}]
+
+  It should be able to handle multiple parameters per key:
+
+      iex> ICalendar.Util.Deserialize.retrieve_params(
+      ...>   "KEY;LOREM=ipsum;DOLOR=sit")
+      ["KEY", %{"LOREM" => "ipsum", "DOLOR" => "sit"}]
+  """
+  def retrieve_params(key) do
+    [key | params] = String.split(key, ";", trim: true)
+
+    params =
+      params
+      |> Enum.reduce(%{}, fn(param, acc) ->
+        [key, val] = String.split(param, "=", parts: 2, trim: true)
+        Map.merge(acc, %{key => val})
+      end)
+
+    [key, params]
   end
 
-  def parse_attr({"DESCRIPTION", description}, acc) do
+  def parse_attr(%Property{key: "DESCRIPTION", value: description},
+                 acc) do
     %{acc | description: desanitized(description)}
   end
-  def parse_attr({"DTSTART", dtstart}, acc) do
-    {:ok, timestamp} = to_date(dtstart)
+  def parse_attr(%Property{key: "DTSTART", value: dtstart, params: params},
+                 acc) do
+    {:ok, timestamp} = to_date(dtstart, params)
     %{acc | dtstart: timestamp}
   end
-  def parse_attr({"DTEND", dtend}, acc) do
-    {:ok, timestamp} = to_date(dtend)
+  def parse_attr(%Property{key: "DTEND", value: dtend, params: params},
+                 acc) do
```

Comment:
@lpil I'm not particularly happy with the way I formatted the `parse_attr/2` arguments to pass dogma. Any suggestions?